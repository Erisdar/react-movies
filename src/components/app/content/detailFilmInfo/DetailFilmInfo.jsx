import FilmImage from 'app/content/filmImage/FilmImage';
import React, { useEffect } from 'react';
import Button from 'react-bootstrap/Button';
import './detailFilmInfo.css';

export default ({ film, setBrieflyMode, showPopularFilms, loadFoundedFilms }) => {

    useEffect(() => {
        window.scrollTo(0, 0);
        loadFoundedFilms({ genre: film.genres[0] });
    })

    return (
        <div className="detail-film-info content__detail-film-info d-flex justify-content-around flex-wrap flex-md-row flex-column align-items-center align-self-center">
            <Button onClick={() => { setBrieflyMode(); showPopularFilms() }} variant="outline-primary" className="ml-auto detail-film-info__search-button">Search</Button>
            <div className="detail-film-info__image">
                <FilmImage logo={film.poster_path} alt={film.title} />
            </div>
            <div className="detail-film-info__info d-flex flex-column text-justify">
                <div className="d-flex font-weight-bold flex-wrap align-items-center justify-content-between">
                    <p className="detail-film-info__title detail-film-info__indent">{film.title}</p>
                    <p className="detail-film-info__rating">{film.vote_average}</p>
                </div>
                <div className="d-flex font-weight-bold detail-film-info__indent">
                    <p>{new Date(film.release_date).getFullYear()}</p>
                    <p>{film.runtime} min</p>
                </div>
                <p className="detail-film-info__indent m-0">{film.overview}</p>
            </div>
        </div>
    );
};