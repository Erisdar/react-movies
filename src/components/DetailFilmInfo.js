import React from 'react';
import '../styles/detailFilmInfo.css';
import FilmImage from './film/FilmImage';
import Button from 'react-bootstrap/Button';

export default ({film, changeView}) => {

    window.scrollTo(0, 0);

    return (
        <>
            <div className="detail-view d-flex justify-content-around flex-wrap flex-md-row flex-column align-items-center align-self-center">
                <Button onClick={() => changeView()} variant="outline-primary" className="ml-auto search-button">Search</Button>
                <div className="film-big-logo">
                    <FilmImage logo={film.logo} alt={film.alt} />
                </div>
                <div className="film-detail-info-container d-flex flex-column text-justify">
                    <div className="d-flex font-weight-bold flex-wrap align-items-center justify-content-between">
                        <p className="film-detailt-title info-text-indent">{film.title}</p>
                        <p className="film-rating">{film.rating}</p>
                    </div>
                    <div className="d-flex font-weight-bold info-text-indent">
                        <p>{film.year}</p>
                        <p>{film.duration} min</p>
                    </div>
                    <p className="info-text-indent m-0">{film.description}</p>
                </div>
            </div>
        </>
    );
}