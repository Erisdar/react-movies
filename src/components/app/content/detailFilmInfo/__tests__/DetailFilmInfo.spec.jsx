import { mount } from 'enzyme';
import React from 'react';
import DetailFilmInfo from '../DetailFilmInfo';

const testFilm = require('data/film.json').find(film => film.id === "1");

describe('DetailFilmInfo component', () => {
    window.scrollTo = jest.fn();

    it('should be render correctly', () => {
        const component = mount(<DetailFilmInfo film={testFilm} />);
        expect(component).toMatchSnapshot();
    });

    it('should call changeView()', () => {
        const changeView = jest.fn();
        const component = mount(<DetailFilmInfo film={testFilm} changeView={changeView} />);
        component.find('button.detail-film-info__search-button').simulate('click');
        expect(changeView).toHaveBeenCalled();
    });
});