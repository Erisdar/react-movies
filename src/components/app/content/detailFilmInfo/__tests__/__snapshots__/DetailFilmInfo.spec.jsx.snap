// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailFilmInfo component should be render correctly 1`] = `
<_default
  film={
    Object {
      "duration": "111",
      "genres": Array [
        "Action",
        "Adventure",
      ],
      "id": "1",
      "overview": "The lead character, called 'The Bride', was a member of the Deadly Viper Assassination Squad, led by her lover 'Bill.' Upon realizing she was pregnant with Bill's child, 'The Bride' decided to escape her life as a killer. She fled to Texas, met a young man, who, on the day of their wedding rehearsal was gunned down by an angry and jealous Bill (with the assistance of the Deadly Viper Assassination Squad). Four years later, 'The Bride' wakes from a coma, and discovers her baby is gone. She, then, decides to seek revenge upon the five people who destroyed her life and killed her baby. The saga of Kill Bill Volume I begins.",
      "poster_path": "kill-bill-v1.jpg",
      "release_date": "2003",
      "title": "Kill Bill vol 1",
      "vote_average": "8.1",
    }
  }
  loadFoundedFilms={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "genre": "Action",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  setBrieflyMode={[MockFunction]}
  showPopularFilms={[MockFunction]}
>
  <div
    className="detail-film-info content__detail-film-info d-flex justify-content-around flex-wrap flex-md-row flex-column align-items-center align-self-center"
  >
    <Button
      active={false}
      className="ml-auto detail-film-info__search-button"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="outline-primary"
    >
      <button
        className="ml-auto detail-film-info__search-button btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </Button>
    <div
      className="detail-film-info__image"
    >
      <_default
        alt="Kill Bill vol 1"
        logo="kill-bill-v1.jpg"
      >
        <img
          alt="Kill Bill vol 1"
          className="film-item__film-image"
          src="kill-bill-v1.jpg"
        />
      </_default>
    </div>
    <div
      className="detail-film-info__info d-flex flex-column text-justify"
    >
      <div
        className="d-flex font-weight-bold flex-wrap align-items-center justify-content-between"
      >
        <p
          className="detail-film-info__title detail-film-info__indent"
        >
          Kill Bill vol 1
        </p>
        <p
          className="detail-film-info__rating"
        >
          8.1
        </p>
      </div>
      <div
        className="d-flex font-weight-bold detail-film-info__indent"
      >
        <p>
          2003
        </p>
        <p>
           min
        </p>
      </div>
      <p
        className="detail-film-info__indent m-0"
      >
        The lead character, called 'The Bride', was a member of the Deadly Viper Assassination Squad, led by her lover 'Bill.' Upon realizing she was pregnant with Bill's child, 'The Bride' decided to escape her life as a killer. She fled to Texas, met a young man, who, on the day of their wedding rehearsal was gunned down by an angry and jealous Bill (with the assistance of the Deadly Viper Assassination Squad). Four years later, 'The Bride' wakes from a coma, and discovers her baby is gone. She, then, decides to seek revenge upon the five people who destroyed her life and killed her baby. The saga of Kill Bill Volume I begins.
      </p>
    </div>
  </div>
</_default>
`;
